src/rules/percent-placeholder-pattern/index.js

```diff
@@ -0,0 +1,70 @@
+import _ from "lodash"
+import resolveNestedSelector from "postcss-resolve-nested-selector"
+import { utils } from "stylelint"
+import {
+  hasInterpolatingAmpersand,
+  isStandardRule,
+  isStandardSelector,
+  parseSelector,
+} from "../../utils"
+
+export const ruleName = "percent-placeholder-pattern"
+
+export const messages = utils.ruleMessages(ruleName, {
+  expected: placeholder => `Expected %-placeholder "%${placeholder}" to match specified pattern`,
+})
+
+export default function (pattern) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: pattern,
+      possible: [ _.isRegExp, _.isString ],
+    })
+    if (!validOptions) { return }
+
+    const placeholderPattern = (_.isString(pattern))
+      ? new RegExp(pattern)
+      : pattern
+
+    // Checking placeholder definitions (looking among regular rules)
+    root.walkRules(rule => {
+      const { selector } = rule
+
+      // If it's a custom prop or a less mixin
+      if (!isStandardRule(rule)) { return }
+      // If the selector has interpolation
+      if (!isStandardSelector(selector)) { return }
+      
+      // Nested selectors are processed in steps, as nesting levels are resolved.
+      // Here we skip processing intermediate parts of selectors (to process only fully resolved selectors)
+      // if (rule.nodes.some(node => node.type === "rule" || node.type === "atrule")) { return }
+
+      // Only resolve selectors that have an interpolating "&"
+      if (hasInterpolatingAmpersand(selector)) {
+        resolveNestedSelector(selector, rule).forEach(selector => {
+          parseSelector(selector, result, rule, s => checkSelector(s, rule))
+        })
+      } else {
+        parseSelector(selector, result, rule, s => checkSelector(s, rule))
```

Comment:
This call is the same as above. It could be extracted into a function:

```js
function parse(selector) {
  parseSelector(selector, result, rule, s => checkSelector(s, rule)
}
resolveNestedSelector(selector, rule).forEach(parse)
```